LOADING_SUPPORT.md

```diff
@@ -0,0 +1,185 @@
+* Bulk loading requires some DBAPI support specifically for this mode.
+  Sometimes there's even a SqlAlchemy Command class that has been
+  cooked up to support.  Read the docs on your DBAPI driver and look
+  to see if there's such a Command class.
+
+## Build a proof of concept using Python drivers
+
+Create a small Python file that tries to load a trivial file into a
+table to show you can get a file off of your local machine and into
+the database.
+
+Along the way, figure out which one of of these your database wants most to do:
+
+* Load from a path on disk or URL (e.g., S3)
+* Load from a file stream (a "fileobj" in Python)
+
+## Create some scaffolding
+
+1. Create an empty subclass of either
+   `records_mover.db.loader.LoaderFromRecordsDirectory` or
+   `records_mover.db.loader.LoaderFromFileobj` in
+   `records_mover.db.your_db_type`, similar to other database types.  You can ignore
+   the MyPy errors for now.
+2. Modify `loader()` in your `DBDriver` subclass to return an instance
+   of your class.  If you are implementing `LoaderFromFileobj`, also
+   have `loader_from_fileobj()` return it.
+3. Using the mypy errors as a driver (run `make` to get mypy to run),
+   create these methods in the DBDriver subclass - initially with
+   `raise NotImplementedError` as the body.  Focus for now at getting
+   the right methods in place and wired up, not yet on adding
+   actual logic.
+
+## Get tests wired up and talking to your code
+
+1. Run the `./itest` target for your database.  You should expect to
+   see a lot of errors from your NotImplementedErrors.  Great!  That
+   means you wired things up correctly to use your loader, which is
+   all we're testing at this point.
+2. Assuming your database uses SQL or maybe a driver function call to
+   initiate bulk loads, this is your chance to figure out how how
+   you're going to generate the different SQL or function call
+   arguments based on what different RecordsFormat options out there.
+
+   Create a function signature describing that approach, in a new file
+   (e.g., `records_mover.db.your_db_type.load_options`) in your Loader
+   class.  Name it (and the file it's in) after the SQL or function call
+   syntax specific to your database - e.g., the Redshift function is named
+   after Redshift's `UNLOAD` statement.
+
+   Thisf unction should just raise `NotImplementedError` for now.
+   Example:
+
+   ```python
+   def redshift_unload_options(unhandled_hints: Set[str],
+                               records_format: BaseRecordsFormat,
+                               fail_if_cant_handle_hint: bool) -> RedshiftUnloadOptions:
+   ```
+
+   In the case above, there was an existing function to do unloading,
+   and `RedshiftUnloadOptions` is a MyPy `TypedDict` describing the
+   keyword arguments to that function.
+
+   If you're constructing SQL yourself, you might consider making a
+   class to do it, perhaps subclassed from `typing.NamedTuple`.  See
```

Comment:
Is there any reason to prefer dataclasses here?  I was reminded that they exist after going down the NamedTuple approach.